From 1348442bd1b70a95a06754407d273cc09981768e Mon Sep 17 00:00:00 2001
From: Nariman Poushin <nariman@opensource.wolfsonmicro.com>
Date: Fri, 14 Feb 2014 05:01:15 +0000
Subject: [PATCH 129/351] mfd: arizona: Move measured hp_impedance to arizona

Make the measured hp_impedance available to the whole arizona codec.

Change-Id: I87e563d99709330aa7d4a0b341afd234ad7613b9
Signed-off-by: Nariman Poushin <nariman@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/switch/switch-arizona.c  |    9 ++++-----
 include/linux/mfd/arizona/core.h |    2 ++
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/switch/switch-arizona.c b/drivers/switch/switch-arizona.c
index e6cf898..91f21e1 100644
--- a/drivers/switch/switch-arizona.c
+++ b/drivers/switch/switch-arizona.c
@@ -76,7 +76,6 @@ struct arizona_extcon_info {
 	struct input_dev *input;
 
 	u16 last_jackdet;
-	u32 hp_impedance;
 
 	int micd_mode;
 	const struct arizona_micd_config *micd_modes;
@@ -499,10 +498,10 @@ static int arizona_hpdet_read(struct arizona_extcon_info *info)
 		}
 	}
 
-	info->hp_impedance = val;
+	arizona->hp_impedance = val;
 
 	if (arizona->pdata.hpdet_cb)
-		arizona->pdata.hpdet_cb(info->hp_impedance);
+		arizona->pdata.hpdet_cb(arizona->hp_impedance);
 
 	dev_dbg(arizona->dev, "HP impedance %d ohms\n", val);
 	return val;
@@ -1246,7 +1245,7 @@ static irqreturn_t arizona_jackdet(int irq, void *data)
 		info->mic = false;
 		info->hpdet_done = false;
 		info->hpdet_retried = false;
-		info->hp_impedance = 0;
+		arizona->hp_impedance = 0;
 		arizona->hp_short = false;
 
 		for (i = 0; i < info->num_micd_ranges; i++)
@@ -1366,7 +1365,7 @@ static ssize_t arizona_extcon_show(struct device *dev,
 	struct platform_device *pdev = to_platform_device(dev);
 	struct arizona_extcon_info *info = platform_get_drvdata(pdev);
 
-	return scnprintf(buf, PAGE_SIZE, "%d\n", info->hp_impedance);
+	return scnprintf(buf, PAGE_SIZE, "%d\n", info->arizona->hp_impedance);
 }
 
 static int arizona_extcon_probe(struct platform_device *pdev)
diff --git a/include/linux/mfd/arizona/core.h b/include/linux/mfd/arizona/core.h
index f313719..1ed2079 100644
--- a/include/linux/mfd/arizona/core.h
+++ b/include/linux/mfd/arizona/core.h
@@ -108,6 +108,8 @@ struct arizona {
 	unsigned int hp_ena;
 	unsigned int hp_short;
 
+	unsigned int hp_impedance;
+
 	struct mutex clk_lock;
 	int clk32k_ref;
 
-- 
1.7.9.5

